senza/components/taupage_auto_scaling_group.py

```diff
@@ -14,6 +14,24 @@
 
 _AWS_FN_RE = re.compile(r"('[{]{2} (.*?) [}]{2}')", re.DOTALL)
 
+# from kio OpenAPI yaml
+APPLICATION_ID_RE = re.compile(r"^[a-z][a-z0-9-]*[a-z0-9]$")
+APPLICATION_VERSION_RE = re.compile(r"^[A-Za-z0-9](?:[A-Za-z0-9._-]*[A-Za-z0-9])?$")
```

Comment:
This RE does have unnecessary stuff also it matches something like ``1..3.....4`` as a valid version.
 I would recommend using this simplified version ``^[A-Za-z0-9]+([._-]?[A-Za-z0-9]+)*$``